hypershift-operator/controllers/nodepool/openstack/openstack.go

```diff
@@ -72,3 +76,62 @@ func MachineTemplateSpec(hcluster *hyperv1.HostedCluster, nodePool *hyperv1.Node
 	}
 	return openStackMachineTemplate, nil
 }
+
+func ReconcileOpenStackImageCR(ctx context.Context, client client.Client, createOrUpdate upsert.CreateOrUpdateFN, hcluster *hyperv1.HostedCluster, release *releaseinfo.ReleaseImage) error {
+	openStackImage := orc.Image{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "rhcos-" + hcluster.Name,
+			Namespace: hcluster.Namespace,
+			// TODO: add proper cleanup in CAPI resources cleanup
+			OwnerReferences: []metav1.OwnerReference{
+				{
+					APIVersion: hcluster.APIVersion,
+					Kind:       hcluster.Kind,
+					Name:       hcluster.Name,
+					UID:        hcluster.UID,
+				},
+			},
+		},
+		Spec: orc.ImageSpec{},
+	}
+
+	if _, err := createOrUpdate(ctx, client, &openStackImage, func() error {
+		err := reconcileOpenStackImageSpec(hcluster, &openStackImage.Spec, release)
+		if err != nil {
+			return err
+		}
+		return nil
+	}); err != nil {
+		return err
+	}
+	return nil
+}
+
+func reconcileOpenStackImageSpec(hcluster *hyperv1.HostedCluster, openStackImageSpec *orc.ImageSpec, release *releaseinfo.ReleaseImage) error {
+	imageURL, imageHash, err := releaseinfo.UnsupportedOpenstackDefaultImage(release)
+	if err != nil {
+		return fmt.Errorf("failed to lookup RHCOS image: %w", err)
+	}
+
+	openStackImageSpec.CloudCredentialsRef = orc.CloudCredentialsReference{
+		SecretName: hcluster.Spec.Platform.OpenStack.IdentityRef.Name,
+		CloudName:  hcluster.Spec.Platform.OpenStack.IdentityRef.CloudName,
+	}
+
+	openStackImageSpec.Resource = &orc.ImageResourceSpec{
+		Name: "rhcos-" + hcluster.Name,
```

Comment:
**FOR CI ONLY**

We'll want an option to override this name so we can re-use the image between CI runs for different clusters. As written this will be expensive to run if the HostedCP is in AWS and the workers are in Vexxhost because we're going to download the bits to AWS before uploading them again to Vexxhost... for each HostedCluster in each CI run. Our shareholders might not be happy with us.

Reply:
Can't we just remain using `openstack-node-image-name` for most of the CI jobs? and some running less frequently without it?